src/firefly/java/edu/caltech/ipac/firefly/server/ws/Workspaces.java

```diff
@@ -54,9 +54,8 @@
      * @return {@link WsResponse}
      * @throws WsException handle ws error
      */
-    WsResponse putFile(String wspaceRelPath, File item, String contentType) throws WsException;
-
-
+    //WsResponse putFile(String wspaceRelPath, File item, String contentType) throws WsException;
+    WsResponse putFile(String wspaceRelPath, boolean writable, File item, String contentType) throws WsException;
```

Comment:
`writable` here is wrongly named.  You also have `overWritable` at other places.
I think it should be either `overwrite` or `shouldOverwrite` and it should be consistently named so that it's easier to read.